src/lib/posthog-client.ts

```diff
@@ -0,0 +1,26 @@
+import posthog from 'posthog-js'
+
+class PosthogClient {
+  static init() {
+    if (typeof window !== 'undefined') {
+      posthog.init(process.env.NEXT_PUBLIC_POSTHOG_KEY || '', {
+        api_host:
+          process.env.NEXT_PUBLIC_POSTHOG_HOST || 'https://app.posthog.com',
+        // Enable debug mode in development
+        loaded: (posthog) => {
+          if (process.env.NODE_ENV === 'development') posthog.debug()
+        },
+        capture_pageview: false, // Disable automatic pageview capture, as we capture manually
+        autocapture: false, // Disable automatic default event capture, as we capture manually
+      })
+    }
+
+    return posthog
+  }
+
+  static capture(event: string, params: any) {
```

Comment:
might consider renaming this `track` to be consistent with our existing usage/apis

does posthog have an analog for `identify` to associate events/sessions with a user?